ceph_deploy/tests/test_cli_mon.py

```diff
@@ -81,9 +79,9 @@ def _create_mon(cluster, get_monitor_secret):
     fake_ip_addresses = lambda x: ['10.0.0.1']
     try:
         with patch('ceph_deploy.new.net.ip_addresses', fake_ip_addresses):
-            with mock.patch('ceph_deploy.new.net.get_nonlocal_ip', lambda x: '10.0.0.1'):
-                with mock.patch('ceph_deploy.new.arg_validators.Hostname', lambda: lambda x: x):
-                    with mock.patch('ceph_deploy.new.hosts'):
+            with patch('ceph_deploy.new.net.get_nonlocal_ip', lambda x: '10.0.0.1'):
```

Comment:
another reason for removal: this nesting is insane. 

Of course I am partly to blame, but these were adapted when poor ceph-deploy didn't have much of tests anyway. The cost of maintenance is far larger than the benefits (for this case and all others that have multiple nested patching)